packages/web/components/MegaMenu.tsx

```diff
@@ -312,31 +318,194 @@ const DesktopNavLinks = () => {
   );
 };
 
-// Search -- not working yet
-// const Search = () => (
-//   <Flex
-//     justifyContent="flex-end"
-//     minW={{ base: '20%', lg: '10%' }}
-//     h="fit-content"
-//     p={2}
-//     mt="auto"
-//     mb="auto"
-//     bg={{ base: 'none', xl: 'rgba(255,255,255,0.05)' }}
-//     border={{ base: 'none', xl: '1px solid #2B2244' }}
-//     borderRadius={4}
-//   >
-//     <Image src={SearchIcon} alt="search" height={16} width={16} />
-//     <Text
-//       display={{ base: 'none', xl: 'block' }}
-//       color="rgba(255,255,255,0.5)"
-//       alignSelf="center"
-//       ml={2}
-//     >
-//       find anything
-//     </Text>
-//   </Flex>
-// );
+export type SearchOption = {
+  value: string;
+  label: string;
+};
+
+const SeeAllResultsComponent = ({
+  text,
+  handleClick,
+}: {
+  text: string;
+  handleClick: () => void;
+}) => (
+  <Box onClick={handleClick}>
+    <Text ml="2" color="#E839B7">
+      See all {text}
+    </Text>
+  </Box>
+);
+
+const MainHeadingComponent = ({ text }: { text: string }) => (
+  <Text fontWeight="bold" color="black">
+    {text}
+  </Text>
+);
+
+const optionsEqualToLimit = (len: number) => len === 3;
+
+const AddSeeAllResultsOption = (
+  options: Array<{ label: JSX.Element; value: string }>,
+  text: string,
+  handleClick: () => void,
+) => [
+  ...options,
+  {
+    label: <SeeAllResultsComponent text={text} handleClick={handleClick} />,
+    value: '',
+  },
+];
+
+const MenuOptionLabel = ({
+  handleClick,
+  name,
+  imgSrc,
+  text,
+}: {
+  handleClick: () => void;
+  name: string;
+  imgSrc?: string;
+  text: string;
+}) => (
+  <Flex align="center" onClick={handleClick}>
+    <Avatar name={name} src={imgSrc} w="20px" h="20px" />
+    <Text ml="2">{text}</Text>
+  </Flex>
+);
 
+const Search = () => {
+  const [inputValue, setValue] = React.useState<string>('');
+  const realtimeInput = React.useRef('');
+  const router = useRouter();
+
+  const debounced = useDebouncedCallback(
+    async () => {
+      const { players } = await getPlayersByText(`%${inputValue}%`);
+      const { guilds } = await getGuildsByText(`%${inputValue}%`);
```

Comment:
The enclosing of the search text with `%`s should happen in the method calling the GraphQL query (i.e. *within* `searchPlayers`).

You just pass in the search string to these methods:

```suggestion
      const { players } = await searchPlayers(search);
      const { guilds } = await searchGuilds(search);
```